onnxruntime/core/providers/rocm/reduction/reduction_ops.cc

```diff
@@ -731,10 +731,9 @@ Status ReduceKernel<allow_multi_axes>::ComputeImpl(OpKernelContext* ctx, miopenR
   std::vector<int64_t> axes;
 
   size_t num_inputs = ctx->InputCount();
-  if (num_inputs == 2) {
+  const Tensor* axes_tensor = num_inputs == 2 ? ctx->Input<Tensor>(1) : nullptr;  // optional input. may be nullptr.
```

Comment:
>const Tensor* axes_tensor [](http://example.com/codeflow?start=2&length=25)

So it is possible to have 2 inputs. but not really?

Reply:
yes. an input may be present with an empty name. It's confusing, but to support multiple optional inputs you may need placeholders like that.